### ISSUE TITLE
Client Frequently hangs gives Timeout error!

### ISSUE THREAD

**Issue by aryan1107** [openai/openai-python#1134]
### Confirm this is an issue with the Python library and not an underlying OpenAI API

- [X] This is an issue with the Python library

### Describe the bug

There are lot of openai.APIConnectionError when there is non english messages sent!

I stream response and have tried with timeout setting `30.0` , `60.0` and even `120.0` 
OpenAI(api_key = os.getenv("OPENAI_API_KEY"),timeout=30.0)

I have noticed there are frequent timeout error and most of the time it's when there is non english messaged but also few times in english messages. Eventually I have to set max retries to 0 and use fallback model.

<img width="864" alt="Screenshot 2023-12-20 at 2 41 47 PM" src="https://github.com/openai/openai-python/assets/31214277/aa2a5c03-9ac5-48a4-b8da-e657490159c0">

### To Reproduce

Essentially  openai.APIConnectionError randomly. I don't even have huge concurrent traffic.

### Code snippets

```Python
#I have tried will all these similar settings while setting streaming = True

client = OpenAI(api_key = os.getenv("OPENAI_API_KEY"),timeout=30.0, max_retries=1)

client = OpenAI(api_key = os.getenv("OPENAI_API_KEY"),timeout=60.0, max_retries=0)

client = OpenAI(api_key = os.getenv("OPENAI_API_KEY"),timeout=80.0, max_retries=0)

```
My OpenAI Library version is always the latest

### OS

Cloud Function

### Python version

Python 3.10

### Library version

v1.11.1

**Comment by aryan1107** [comment]
Update I changed to AzureOpenAI still noticed the same issue after reeving timeouts I will make some changes to timeout settings and it gets okay for the time then again I will start receiving timeout errors.

When I make changes memory utilization also changes.

<img width="866" alt="Screenshot 2024-02-08 at 12 06 07 AM" src="https://github.com/openai/openai-python/assets/31214277/e4088b29-e20c-4107-b450-7288f8c75a4e">

<img width="853" alt="Screenshot 2024-02-08 at 12 08 57 AM" src="https://github.com/openai/openai-python/assets/31214277/a7ccb156-2bfb-460c-85c4-fdb5d86ac0f2">

**Comment by rattrayalex** [comment]
What is your cloud provider? Is it possible there's a networking issue there? What happens if you make the requests directly from the same machines, without the SDK?

**Comment by rattrayalex** [comment]
Does the issue happen with both English and other languages, or only in other languages? If so what other languages?

**Comment by aryan1107** [comment]
@rattrayalex Hi it happens in english too, but other languages like Spanish, Swedish, French are more conman. As per cloud provider it's GCP without SDK I haven't tried but I did try using OpenRouter OpenAI proxy it worked without any issues and timeout error was almost gone. When using OpenRouter API just had to add the base url in sdk however model was same.

This is not just issue with me I found another one faceing similar issue [here](https://community.openai.com/t/api-frequently-hangs-gives-timeout-error/618658/2?u=aryanchaurasia.348)
